core/include/vecmem/concepts/atomic_ref.hpp

```diff
@@ -0,0 +1,62 @@
+/*
+ * VecMem project, part of the ACTS project (R&D line)
+ *
+ * (c) 2022-2023 CERN for the benefit of the ACTS project
+ *
+ * Mozilla Public License Version 2.0
+ */
+#pragma once
+
+#if __cpp_concepts >= 201907L
+#include <concepts>
+
+#include "vecmem/memory/device_atomic_ref.hpp"
```

Comment:
You should move `vecmem::memory_order` into a separate header. As I don't like this circular include.